src/cloud-foundry-api-client.js

```diff
@@ -19,6 +19,48 @@ class CloudFoundryAPIClient {
       )).then(body => this._filterAppsResponse(JSON.parse(body)));
   }
 
+  fetchAppStats(appGUID) {
+    return this._authClient.accessToken().then(token => this._request(
+        'GET',
+        `/v2/apps/${appGUID}/stats`,
+        token
+      ));
+  }
+
+  fetchAppInstanceStates(container) {
+    return this.fetchAppStats(container.guid)
+      .then(stats =>
+        ({
+          guid: container.guid,
+          name: container.name,
+          states: this._appInstanceStates(JSON.parse(stats)),
+        })
+      );
+  }
+
+  fetchAllAppInstanceErrors(buildContainers) {
+    const instanceErrors = [];
+    const promises = [];
+    let states;
+    let i;
+    for (i = 0; i < buildContainers.length; i += 1) {
```

Comment:
It may be a bit cleaner here if we do something like this here:

```javascript
const promises = buildContainers.map(buildContainer => {
  // ...
});
```